packages/contracts/src/tokens/RewardHolder.sol

```diff
@@ -83,26 +86,39 @@ contract RewardHolder is Initializable, AccessControlUpgradeable, ReentrancyGuar
         }
 
         // calculate reward for token owner
-        uint256 tokenOwnerReward = calcReward();
-        rewardOwnerIndex[msg.sender] = rewardIndex;
+        (uint256 tokenOwnerReward, uint256 claimableIndex) = previewReward(msg.sender);
+        if(tokenOwnerReward == 0) {
+            revert CallerHaveZeroReward();
+        }
+        rewardOwnerIndex[msg.sender] = claimableIndex;
 
         // transfer reward to token owner
         bool success = vault.transfer(msg.sender, tokenOwnerReward);
         emit RewardClaimed(tokenOwnerReward, address(msg.sender), success);
     }
 
-    function calcReward() public view returns (uint256) {
-        if(numberCoins == 0 || rewardOwnerIndex[msg.sender] == 0) {
-            return 0;
+    /// @dev calculate reward for token owner and last claimable index
+    function previewReward(address owner) public view returns (uint256, uint256) {
+        if(ownersCount == 0 || rewardOwnerIndex[owner] == 0) {
+            return (0, rewardIndex);
         }
-        uint256 deltaIndex = rewardIndex - rewardOwnerIndex[msg.sender];
-        return deltaIndex / numberCoins;
+        
+        uint256 deltaIndex = rewardIndex - rewardOwnerIndex[owner];
+        // Division rounds down to the nearest integer
+        // As a result we exclude the remainder from the owner index
+        // So he will be able to claim left over reward in the future
+        return (deltaIndex / ownersCount, rewardIndex - deltaIndex % ownersCount);
```

Comment:
tbh i cannot understand this part, why we cannot use the suggestion from audit here? (i.e. increase precision). Can we make a call to discuss this?

Reply:
Yep, discusses in call